azurelinuxagent/common/osutil/redhat.py

```diff
@@ -117,11 +117,38 @@ def set_hostname(self, hostname):
             logger.warn("[{0}] failed, attempting fallback".format(' '.join(hostnamectl_cmd)))
             DefaultOSUtil.set_hostname(self, hostname)
 
+    def get_nm_controlled(self):
+        ifname = self.get_if_name()
+        filepath = "/etc/sysconfig/network-scripts/ifcfg-{0}".format(ifname)
+        nm_controlled_cmd = ['grep', 'NM_CONTROLLED=', filepath]
+        try:
+            result = shellutil.run_command(nm_controlled_cmd, log_error=False).rstrip()
+
+            if result and len(result.split('=')) > 1:
+                value = result.split('=')[1]
+                if "n" in value:
+                    return False
+        except shellutil.CommandError as e:
+            # Command might fail because NM_CONTROLLED value is not in interface config file (exit code 1).
+            # Log warning for any other exit code.
+            # NM_CONTROLLED=y by default if not specified.
+            if e.returncode != 1:
+                logger.warn("[{0}] failed: {1}.\nAgent will continue to publish hostname with NM_CONTROLLED=y".format(' '.join(nm_controlled_cmd), e))
```

Comment:
users might not know about NM_CONTROLLED=y. Do you think logging this "Agent will continue to publish hostname without NetworkManager restart" is clearer?

Reply:
or adding both?